SubEthaEdit-Mac/Source/PlainTextDocument.m

```diff
@@ -6354,12 +6354,25 @@ - (BOOL)textView:(NSTextView *)aTextView shouldChangeTextInRange:(NSRange)aAffec
     NSArray *plainTextEditors = [self plainTextEditors];
     unsigned editorCount = [plainTextEditors count];
     if ([plainTextEditors count] > 1) {
-         I_currentTextOperation = [TextOperation textOperationWithAffectedCharRange:aAffectedCharRange replacementString:aReplacementString userID:(NSString *)[TCMMMUserManager myUserID]];
-        while (editorCount--) {
-            PlainTextEditor *editor = [plainTextEditors objectAtIndex:editorCount];
-            if ([editor textView] != aTextView) {
-                [editor storePosition];
+        TCMMMOperation *currentOperation = [TextOperation textOperationWithAffectedCharRange:aAffectedCharRange replacementString:aReplacementString userID:(NSString *)[TCMMMUserManager myUserID]];
+        
+        if (!textStorage.isBlockediting) {
+            // This method will be called within a -beginEditing/-endEditing section if
+            // block editing is active. In this case, the position will not be stored as
+            // it's not possible to ask the layoutmanager for the glyph index while the
+            // storage is being mutated.
+            // The changes will be added to the I_currentTextOperations array.
+            
+            I_currentTextOperations = [NSMutableArray arrayWithObject:currentOperation];
```

Comment:
hmmm - isn't the real fix here not just to guard against blockediting? The array part doesn't seem to be used at all. I think to not store things while blockediting seems like an okay fix, but I wonder if that covers all the cases where this crashes because we query layout at inopportune times.

Reply:
Oh, had a closer look and unsterood the patch fully. Seems good. Will test and probably pull in. Thanks!